File: src/系统/11．剧情系统/01．主线任务/02．剧情步骤/00．主线剧情/06．沙漠与线索.lua
local ____lualib = require("lualib_bundle")
local __TS__StringIncludes = ____lualib.__TS__StringIncludes
local __TS__StringSplit = ____lualib.__TS__StringSplit
local ____exports = {}
---
-- @noSelfInFile
local jass = require("jass.common")
local jglobals = require("jass.globals")
local ____require_result_0 = require("lib.扩展函数.YDWE函数.09．YDUserData安全版")
local YDUserDataGetSafe = ____require_result_0.YDUserDataGetSafe
do
    local ____06_FF0E_6C99_6F20_4E0E_7EBF_7D22 = require("系统.11．剧情系统.01．主线任务.02．剧情步骤.01．第一章.06．沙漠与线索")
    ____exports["沙漠入口调查剧情片段"] = ____06_FF0E_6C99_6F20_4E0E_7EBF_7D22["沙漠入口调查剧情片段"]
    ____exports["沙漠年轻佣兵线索剧情片段"] = ____06_FF0E_6C99_6F20_4E0E_7EBF_7D22["沙漠年轻佣兵线索剧情片段"]
    ____exports["沙漠年长者线索剧情片段"] = ____06_FF0E_6C99_6F20_4E0E_7EBF_7D22["沙漠年长者线索剧情片段"]
    ____exports["沙漠情报商人线索剧情片段"] = ____06_FF0E_6C99_6F20_4E0E_7EBF_7D22["沙漠情报商人线索剧情片段"]
end
local IssueImmediateOrder = jass.IssueImmediateOrder
local Player = jass.Player
local RemoveDestructable = jass.RemoveDestructable
local SetUnitOwner = jass.SetUnitOwner
local function _____8BFB_53D6_4E3B_7EBFNPC(_____5F15_7528)
    local _____952E_540D = __TS__StringIncludes(_____5F15_7528, ".") and (__TS__StringSplit(_____5F15_7528, ".")[2] or "") or _____5F15_7528
    if _____952E_540D == "" then
        return nil
    end
    return YDUserDataGetSafe("string", "主线NPC", _____952E_540D, "unit")
end
local function _____6267_884C_7EBF_7D22NPC_5BF9_8BDD_524D_7F6E(_____53C2_6570)
    local ____8BFB_53D6_4E3B_7EBFNPC_2 = _____8BFB_53D6_4E3B_7EBFNPC
    local ____53C2_6570_NPC_1 = _____53C2_6570.NPC
    if ____53C2_6570_NPC_1 == nil then
        ____53C2_6570_NPC_1 = ""
    end
    local npc = ____8BFB_53D6_4E3B_7EBFNPC_2(tostring(____53C2_6570_NPC_1))
    if npc ~= nil and npc ~= 0 then
        SetUnitOwner(
            npc,
            Player(6),
            true
        )
    end
    local _____89E6_53D1_5355_4F4D = YDUserDataGetSafe("string", "主线剧情入口", "触发单位", "unit")
    if _____89E6_53D1_5355_4F4D ~= nil and _____89E6_53D1_5355_4F4D ~= 0 then
        IssueImmediateOrder(_____89E6_53D1_5355_4F4D, "stop")
    end
end
____exports["执行沙漠情报商人对话前置"] = function(_____53C2_6570)
    _____6267_884C_7EBF_7D22NPC_5BF9_8BDD_524D_7F6E(_____53C2_6570)
    local ____53C2_6570__7834_574F_7269_3 = _____53C2_6570["破坏物"]
    if ____53C2_6570__7834_574F_7269_3 == nil then
        ____53C2_6570__7834_574F_7269_3 = ""
    end
    local _____7834_574F_7269_540D = tostring(____53C2_6570__7834_574F_7269_3)
    if _____7834_574F_7269_540D == "" then
        return
    end
    local destructable = jglobals[_____7834_574F_7269_540D]
    if destructable ~= nil and destructable ~= 0 then
        RemoveDestructable(destructable)
    end
end
____exports["沙漠与线索剧情动作注册表"] = {["JLC沙漠_年轻佣兵对话前置"] = _____6267_884C_7EBF_7D22NPC_5BF9_8BDD_524D_7F6E, ["JLC沙漠_年长者对话前置"] = _____6267_884C_7EBF_7D22NPC_5BF9_8BDD_524D_7F6E, ["JLC沙漠_情报商人对话前置"] = ____exports["执行沙漠情报商人对话前置"]}
return ____exports

File: TS/系统/11．剧情系统/01．主线任务/02．剧情步骤/00．主线剧情/06．沙漠与线索.ts
/** @noSelfInFile */

const jass = require("jass.common") as any;
const jglobals = require("jass.globals") as any;

const { YDUserDataGetSafe } = require("lib.扩展函数.YDWE函数.09．YDUserData安全版") as {
  YDUserDataGetSafe: (this: void, tableType: string, tableKey: any, attr: string, valueType: string) => any;
};

import type { 剧情动作参数表, 剧情动作处理器 } from "../../00．剧情系统核心工具/00．剧情动作类型";
import { 给玩家组添加区域视野 } from "../../00．剧情系统核心工具/06．剧情通用执行工具";
export {
  沙漠入口调查剧情片段,
  沙漠年轻佣兵线索剧情片段,
  沙漠年长者线索剧情片段,
  沙漠情报商人线索剧情片段,
} from "../01．第一章/06．沙漠与线索";

const IssueImmediateOrder = jass.IssueImmediateOrder as (this: void, whichUnit: any, order: string) => boolean;
const Player = jass.Player as (this: void, whichPlayer: number) => any;
const RemoveDestructable = jass.RemoveDestructable as (this: void, whichDestructable: any) => void;
const SetUnitOwner = jass.SetUnitOwner as (this: void, whichUnit: any, whichPlayer: any, changeColor: boolean) => void;

function 读取主线NPC(this: void, 引用: string): any {
  const 键名 = 引用.includes(".") ? 引用.split(".")[1] ?? "" : 引用;
  if (键名 === "") return null;
  return YDUserDataGetSafe("string", "主线NPC", 键名, "unit");
}

function 执行线索NPC对话前置(this: void, 参数: 剧情动作参数表): void {
  const npc = 读取主线NPC(String(参数.NPC ?? ""));
  if (npc != null && npc !== 0) {
    SetUnitOwner(npc, Player(6), true);
  }
  const 触发单位 = YDUserDataGetSafe("string", "主线剧情入口", "触发单位", "unit");
  if (触发单位 != null && 触发单位 !== 0) {
    IssueImmediateOrder(触发单位, "stop");
  }
}

export function 执行沙漠情报商人对话前置(this: void, _参数: 剧情动作参数表): void {
  const 触发单位 = YDUserDataGetSafe("string", "主线剧情入口", "触发单位", "unit");
  if (触发单位 != null && 触发单位 !== 0) {
    IssueImmediateOrder(触发单位, "stop");
  }
}

export function 执行情报商人线索显视野(this: void, 参数: 剧情动作参数表): void {
  const 视野矩形 = String(参数.视野矩形 ?? "");
  if (视野矩形 !== "") 给玩家组添加区域视野(视野矩形);
}

export function 执行情报商人线索清理阻挡物(this: void, 参数: 剧情动作参数表): void {
  const 破坏物名 = String(参数.破坏物 ?? "");
  if (破坏物名 === "") return;
  const destructable = jglobals[破坏物名];
  if (destructable != null && destructable !== 0) {
    RemoveDestructable(destructable);
  }
}

export const 沙漠与线索剧情动作注册表: Record<string, 剧情动作处理器> = {
  "JLC沙漠_年轻佣兵对话前置": 执行线索NPC对话前置,
  "JLC沙漠_年长者对话前置": 执行线索NPC对话前置,
  "JLC沙漠_情报商人对话前置": 执行沙漠情报商人对话前置,
  "JLC沙漠_情报商人线索显视野": 执行情报商人线索显视野,
  "JLC沙漠_情报商人线索清理阻挡物": 执行情报商人线索清理阻挡物,
};
